rs/ic_os/partition_tools/src/ext.rs

```diff
@@ -230,6 +206,48 @@ impl ExtPartition {
 
         Ok(())
     }
+
+    fn check_debugfs_result(out: &Output) -> Result<()> {
+        let errors = std::str::from_utf8(&out.stderr)?
+            .lines()
+            .filter(|error| !error.starts_with("debugfs")) // version number
+            // Some errors we ignore.
+            .filter(|error| !error.contains("Ext2 directory already exists"))
+            .filter(|error| !error.contains("rm: File not found"))
+            .join("\n");
+        if !out.status.success() || !errors.is_empty() {
+            bail!("debugfs failed:\n{errors}");
+        }
+
+        Ok(())
+    }
+
+    fn debugfs_input(input: &Path, output: &Path) -> String {
+        // Commands that generate all parent dirs of input (debugfs doesn't
+        // support mkdir -p)
+        // Eg. if `output` is `/opt/ic/bin/something.txt`, `mkdir_all` will be:
+        // mkdir /opt
+        // mkdir /opt/ic
+        // mkdir /opt/ic/bin
+        let mkdir_all = std::iter::successors(output.parent(), |child| child.parent())
```

Comment:
Was this failing at all in the current build?

Reply:
I didn't check, if you want, I can temporarily revert the mkdir change and just run the error handling change to see if anything changes. I believe we only copy into dirs that are created earlier in the process.

I wanted to see if we can move some copying from the Dockerfile into inject_files to speed up the build and the component existence test that I added last week started to fail. I had to debug it for a while until I found out that inject_files was silently skipping those files.

Reply:
If it is easy, I think it would be good to check, that way we can check in if we've been skipping errors or missing anything 😱.

Reply:
I checked and the existing build seems fine.